### ISSUE TITLE
Please add description how to iterate over a map into the Readme

### ISSUE THREAD

**Issue by slavaschmidt** [hungrybluedev/whisker#5]
### Describe the feature

Please add description how to iterate over a map into the Readme

### Use Case

Currently, the Readme describes iteration over items of an array and using a map with hard-coded key values.
It would be great to have an example of how to iterate over a map using dynamic key/value pairs.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

0.2.0

### Environment details (OS name and version, etc.)

any

**Comment by hungrybluedev** [comment]
Iteration over map's key value pairs isn't supported at the moment. However, you can use a list of maps to achieve what you are looking for.

Check the spec directory for examples.

If you still want this feature, please add a detailed input, template and expected output. I can then use it as a reference to implement the requested feature.

**Comment by slavaschmidt** [comment]
Thank you for the suggestion about the list of maps. I understand converting any map into a list of maps should be possible by converting a value into a map and moving the key into it.

Still, having the possibility to iterate over a map could save this need for transformation and might be useful. For example, consider having a map with a key, the name of a sensor, and the value is an array of measurements. Converting it into an array of maps would add unneeded complexity.

I think it is impossible to solve the key/value requirement without introducing an additional synthetic variable, say "key".

Could something like this work?

```
json-data: {"map" : {"a": "1", "b": "2", "c": "3" } }
template: '"{{*map}}({{.key}}:{{.}}){{/map}}"'
expected: '"(a:1)(b:1)(c:1)"'
```

I also have a completely different question: is it possible to check arrays for emptiness? For example, to only render the head of the table if it is known to contain at least one row.

Thank you!

**Comment by hungrybluedev** [comment]
1. The example is a good starting point. I'll see what I can add next week to support iteration on key value pairs.
2. Use an extra Boolean value in the data model to indicate the absence/presence of data and render the header based on that. This is a deliberate choice to keep only one way of doing things and keeping the template engine simple.

Note that I'm assuming that you have control over the data model before you pass it to the whisker template renderer. Therefore, adding one more Boolean value like `has_items: data.len > 0` and then using that in the template should be relatively simple.

**Comment by slavaschmidt** [comment]
1. Cool, thanks a lot
2. I'm already doing this with slightly different syntax (and also converting all non-strings into strings, for HTML the type does not matter). My reasoning was that this is such a standard requirement for HTML generation that it might be a good idea to include it in the default behavior.

Many thanks!

**Comment by hungrybluedev** [comment]
Can you please collect the summary of the discussion on list length/emptyness discussion and create a separate issue?

Will treat that as a separate feature request.

**Comment by hungrybluedev** [comment]
Supporting this feature will require a bit more time and thought. I'll try implementing the extension of position and negative section support for lists (and then maps) first.

**Comment by hungrybluedev** [comment]
I've given it some thought and I think I have found a way to proceed. I'll try to implement this feature this week.

**Comment by hungrybluedev** [comment]
Implemented in daa93535cb0060bb9379520e80de61cf974f68d7.

@slavaschmidt Can you add examples to the README and the unit test like last time?

**Comment by slavaschmidt** [comment]
@hungrybluedev sorry, I am currently under a heavy workload. I'll try to find some time in a few days.
